### ISSUE TITLE
[alpha] Configuring Gateway Network Topology

### ISSUE THREAD

**Issue by jacob-delgado** [istio/enhancements#20]
Configuring Gateway Network Topology is currently experimental. This is the checklist required to make sure that it can be elevated to Alpha.

The document can be found here: https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/

This document outlines the requirements outlined at https://github.com/istio/community/edit/master/FEATURE-LIFECYCLE-CHECKLIST.md

## Alpha feature requirements

A development feature can be officially labeled as Alpha once it meets the following
requirements:

- [x] Design
  - [x] Requires Approved RFC. - Not needed as it was approved in 1.6 and was approved in istio/api. Design only affects meshConfig which means it cannot be promoted past Beta anyway.

- [x] Config
  - [x] Requires explicit user action to enable (e.g. a config field, config resource, or installation action).
    - [x] istio api: https://github.com/istio/api/pull/1394 and https://github.com/istio/api/pull/1403
    - [x] istio/istio: https://github.com/istio/istio/pull/23292

- [x] Docs
  - [x] Basic feature docs are published on istio.io describing what the feature does, how to use it, and any caveats.
    - [x] See https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/

- [ ] Tests
  - [x] Automated integration tests cover core use cases with the feature enabled. See https://github.com/istio/istio/pull/30500 and https://github.com/istio/istio/pull/27861
  - [x] When disabled the feature does not affect system stability or performance.
  - [ ] Istio.io automated tests

- [ ] Performance
  - [ ] Performance impacts have been measured.

- [x] API
   - [x] Initial API review.

**Comment by linsun** [comment]
I approve pending tests are automated in istio.io

**Comment by smawson** [comment]
I'm good with this going to alpha once tests are in place, but I have some comments:
- The docs should explain a little bit earlier why a user would want to use this feature. It assumes the reader knows that they want these features and is trying to implement them, but I think we could also provide some more context.
- XFCC doc should provide guidance on why a user would want to use one of the values, and not just delegate to Envoy docs. We should tell the user what the feature is for and why (and how) they should use it.
- For PROXY protocol, we need to move towards a mechanism that doesn't rely on an EnvoyFilter. Fine for Alpha I think but not for Beta or beyond. Seems like it could go in gatewayTopology as well.

So basically I really want us to provide guidance to users, not just features.

**Comment by brian-avery** [comment]
Migrated to https://github.com/brian-avery/enhancements/blob/migrateFeaturesToFiles/features/configuring_gateway_network_topology.md as part of #26
